Address derivation and portability (#818)
w3f/polkadot-wiki: docs/learn-accounts.md

It's important to understand that the different formats for different networks are merely different
representations of the same public key in a private-public keypair generated by an address

> generation tool. This makes the addresses cross-Substrate-chain-compatible as long as you convert

Reviewer: ```suggestion
generation tool. This makes the addresses compatible across Substrate-based chains as long as you convert
```

the format.

## Address Generation, Derivation, and Portability

A valid account only requires a private key that can sign on one of the
[supported curves and signature schemes](build-protocol-info#cryptography). Most wallets take many
steps from a mnemonic phrase to account key, which affects the ability to use the same mnemonic
phrase in multiple wallets. Wallets that use different steps will arrive at a different set of
addresses from the same mnemonic.

### Seed Generation

Most wallets generate a mnemonic phrase for users to back up their wallets and generate a private
key from the mnemonic. Not all wallets use the same algorithm to convert from mnemonic phrase to
private key.

> Subkey and Polkadot JS based wallets use the BIP39 dictionary for mnemonic generation, but use the

Reviewer: ```suggestion
Subkey and Polkadot-JS based wallets use the BIP39 dictionary for mnemonic generation, but use the
```

entropy byte array to generate the private key, while full BIP39 wallets (like Ledger) use 2048
rounds of PBKDF2 on the mnemonic. As such, the same mnemonic will not generate the same private
keys. See [Substrate BIP39](https://github.com/paritytech/substrate-bip39) for more information.

### Derivation Paths

Many Polkadot key generation tools support hard and soft derivation. See the
[Subkey documentation](https://substrate.dev/docs/en/knowledgebase/integrate/subkey) for details and

> examples of derivation path formats. The Polkadot JS Apps and Extension as well as Parity Signer

Reviewer: ```suggestion
examples of derivation path formats. The Polkadot-JS Apps and Extension as well as Parity Signer
```

unless both wallets use the same derivation paths.

Polkadot and Kusama both have paths registered in the
[BIP44 registry](https://github.com/satoshilabs/slips/blob/master/slip-0044.md).

> **Warning:** You must have both the parent private key as well as the derivation path to arrive at
> the key for an address. Do not use custom derivation paths unless you are comfortable with your
> understanding of this topic.

### Portability

The above information brings us to portability: the ability to use a mnemonic phrase or seed across
multiple wallets. Portability depends on a number of factors:

- Derivation path
- Mnemonic format
- Seed derivation
- Signature scheme

If you want to use the same mnemonic across multiple wallets, make sure that they follow compatible
methods for generating keys and signing messages. If you cannot find understandable documentation,
reach out to the project maintainers.

|                       | Mnemonic Format | Derivation Path | Seed Derivation |      Signature Support      |
| :-------------------- | :-------------: | :-------------: | :-------------: | :-------------------------: |

> | Polkadot JS Extension |    Standard     |  User-Defined   |      BIP32      |           sr25519           |

Reviewer: ```suggestion
| Polkadot-JS Extension |    Standard     |  User-Defined   |      BIP32      |           sr25519           |
```

> | Polkadot JS Apps      |   Standard\*    |  User-Defined   |      BIP32      | sr25519, ed25519, secp256k  |

Reviewer: ```suggestion
| Polkadot-JS Apps      |   Standard\*    |  User-Defined   |      BIP32      | sr25519, ed25519, secp256k  |
```
